### ISSUE TITLE
Efficient implementation of the Boosting algorithm

### ISSUE THREAD

**Issue by raharth** [raharth/PyTorch_Lib#50]
In the original paper "Deep Exploration via Bootstrapped DQN" they claim that they can efficiently train a boosted network with an approach similar to dropout, which I honestly dont fully gasp.

There might be a neat way of doing it though by incorporating the core into the dataloader, which forwards the entire dataset once, sampling from the output of the core. We then only need the core module to be added to the parameters of the optimizer, or have an optimizer itself.

In the original version additional data is also sampled only from one single head not all of them. This is not a major modification though, since I fill the buffer with all heads but sampling less data from each of them.

**Comment by raharth** [comment]
We will not proceed using boosting therefore not longer necessary
